cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/BasicWorkflowToken.java

```diff
@@ -117,6 +121,11 @@ public void put(String key, Value value) {
   }
 
   void put(String key, Value value, Scope scope) {
+    // Check if put operation is being performed inside the Spark executor
+    if (!putOperationAllowed) {
+      throw new UnsupportedOperationException("Put operation is not allowed from Spark executor.");
```

Comment:
Shouldn't mention Spark in this class, since it is not spark specific. E.g. in Mapper/Reducer context, put is also not allowed.